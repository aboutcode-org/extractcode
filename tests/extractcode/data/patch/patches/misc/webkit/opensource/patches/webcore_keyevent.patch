diff -ur src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp~
--- src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp	2009-12-08 14:32:54.461258995 +0800
+++ src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp~	2009-12-08 14:32:54.425258494 +0800
@@ -1592,7 +1592,13 @@
             defaultTabEventHandler(event);
    }
    if (event->type() == keypressEvent) {
-        m_frame->editor()->handleKeyboardEvent(event);
+        //filter out TAB key press event to prevent input TAB in text field, because it should be handled as advance focus
+        if (event->charCode() != 9 /* TAB's ASCII value */) {
+            m_frame->editor()->handleKeyboardEvent(event);
+        }
+        else {
+            event->setDefaultHandled();
+        }
         if (event->defaultHandled())
             return;
    }

