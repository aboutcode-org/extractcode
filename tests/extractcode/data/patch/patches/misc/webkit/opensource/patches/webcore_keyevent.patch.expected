[('src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp',
  'src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp~',
  "diff -ur src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp~\n--- src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp\n+++ src/3rdParty/webkit/work/WebCore/page/EventHandler.cpp~\n@@ -1592,7 +1592,13 @@\ndefaultTabEventHandler(event);\n}\nif (event->type() == keypressEvent) {\n-        m_frame->editor()->handleKeyboardEvent(event);\n+        //filter out TAB key press event to prevent input TAB in text field, because it should be handled as advance focus\n+        if (event->charCode() != 9 /* TAB's ASCII value */) {\n+            m_frame->editor()->handleKeyboardEvent(event);\n+        }\n+        else {\n+            event->setDefaultHandled();\n+        }\nif (event->defaultHandled())\nreturn;\n}")]